packages/components/addon/components/hds/table/th-selectable.js

```diff
@@ -0,0 +1,58 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+import Component from '@glimmer/component';
+import { action } from '@ember/object';
+import { guidFor } from '@ember/object/internals';
+import { tracked } from '@glimmer/tracking';
+
+export default class HdsTableThSelectableComponent extends Component {
+  @tracked _checked = this.checked;
+
+  /**
+   * Generate a unique ID for the Checkbox
+   * @return {string}
+   */
+  checkboxId = 'checkbox-' + guidFor(this);
+
+  get ariaLabel() {
+    if (this.args.scope === 'col') {
+      // TODO: Doesn't update in reaction to row checkboxes being selected
+      if (this._checked === true) {
+        return 'deselect all';
+      } else {
+        return 'select all';
+      }
+    } else {
+      return `Select ${this.args.selectionKey}`;
+    }
+  }
```

Comment:
@MelSumner This is still a work in progress but does this make sense for the checkbox aria-labels?

Reply:
resolving as outdated